extensions/azurecore/src/azureResource/providers/azuremonitor/azuremonitorTreeDataProvider.ts

```diff
@@ -48,7 +48,7 @@ export class AzureMonitorTreeDataProvider extends ResourceTreeDataProviderBase<a
 				groupFullName: '',
 				groupId: '',
 				providerName: 'LOGANALYTICS',
-				saveProfile: false,
+				saveProfile: true,
```

Comment:
It doesn't add connection to 'Server tree' when doing other context actions - as connection is temporary for that.
This fixes the behavior for Connect icon or 'Connect' menu option.

Reply:
So I looked at the command and we're already setting saveProfile to true there : 

https://github.com/Microsoft/azuredatastudio/blob/9708b470c780c227e3ec74ac257d9652dcd8efee/extensions/azurecore/src/azureResource/commands.ts#L188

So I don't see any changes needing to be made on the azurecore side. You just need to fix the main thread side to respect the property correctly - which you've identified being a problem with your other change there. 

And for that it seems like we should be using `default` or `temporary` in `$openConnectionDialog` - having it be "editor" doesn't make sense to me since that should never be called from an editor context since it's coming from an extension. I tried changing that locally and it seems to fix the issue so I'd suggest starting there and seeing if you can find out why we have it set to the editor in the first place (possibly something around CMS, I see some commits involving that stuff with some of these setting overrides)

Reply:
Turns out it does seem to work fine (can't catch any obvious issues) - I did see it was added here.. https://github.com/microsoft/azuredatastudio/pull/5542/files#diff-e39e5be8a469b6510ff10b765cba3965b164961b048215fce178b4dfc7479da8R63-R64

But not sure if changing it to `default` now has any side-effects..
Any concerns @alanrenmsft ?